domain/ide/workspace/folder.go

```diff
@@ -226,7 +213,45 @@ func (f *Folder) processResults(scanData snyk.ScanData) {
 	f.FilterAndPublishCachedDiagnostics(scanData.Product)
 }
 
+func incrementSeverityCount(scanData *snyk.ScanData, issue snyk.Issue) {
+	issueProduct := issue.Product
+	if issueProduct == "" {
+		log.Debug().Str("method", "incrementSeverityCount").Msg("Issue product is empty. Setting to unknown")
+		issueProduct = "unknown"
+	}
+
+	initializeSeverityCountForProduct(scanData, issueProduct)
+
+	switch issue.Severity {
+	case snyk.Critical:
+		scanData.SeverityCount[issueProduct].Critical++
+	case snyk.High:
+		scanData.SeverityCount[issueProduct].High++
+	case snyk.Medium:
+		scanData.SeverityCount[issueProduct].Medium++
+	case snyk.Low:
+		scanData.SeverityCount[issueProduct].Low++
+	}
+}
+
+func initializeSeverityCountForProduct(scanData *snyk.ScanData, productType product.Product) {
+	if scanData.SeverityCount == nil {
+		scanData.SeverityCount = make(map[product.Product]*snyk.SeverityCount)
+	}
+
+	if productType == "" {
+		log.Debug().Str("method", "initializeSeverityCountForProduct").Msg("Product is empty. Setting to unknown")
+		productType = "unknown"
+	}
+
+	if _, exists := scanData.SeverityCount[productType]; !exists {
+		scanData.SeverityCount[productType] = &snyk.SeverityCount{}
+	}
+}
```

Comment:
As we can see from the tests, a `Product` could be defined at `ScanData` or `Issue` level.
- When `incrementSeverityCount` is called from `processResults` in the loop of Issues, we use the `Product` defined at the Issue level.
- When `processResults` is called with a scanData where `Issues` is an empty slice, we use `Product` from `scanData`

I'm not sure if this is the correct behaviour. Please advice :)

Reply:
It is correct :)